drivers/sensor/lm35/lm35.c

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2024 Renato Soma <renatoys08@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT lm35
+
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/adc.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+
+#define LM35_GAIN       ADC_GAIN_1
+#define LM35_REF        ADC_REF_INTERNAL
+
+LOG_MODULE_REGISTER(LM35, CONFIG_SENSOR_LOG_LEVEL);
+
+struct lm35_data {
+	uint16_t raw;
+};
+
+struct lm35_config {
+	const struct device *adc;
+	uint8_t adc_channel;
+	struct adc_sequence adc_seq;
+	struct adc_channel_cfg ch_cfg;
+};
+
+static int lm35_sample_fetch(const struct device *dev, enum sensor_channel chan)
+{
+	if (chan != SENSOR_CHAN_ALL && chan != SENSOR_CHAN_AMBIENT_TEMP) {
+		return -ENOTSUP;
+	}
+
+	const struct lm35_config *cfg = dev->config;
+
+	return adc_read(cfg->adc, &cfg->adc_seq);
+}
+
+static int lm35_channel_get(const struct device *dev, enum sensor_channel chan,
+			    struct sensor_value *val)
+{
+	if (chan != SENSOR_CHAN_ALL && chan != SENSOR_CHAN_AMBIENT_TEMP) {
+		return -ENOTSUP;
+	}
+
+	int err;
+	struct lm35_data *data = dev->data;
+	const struct lm35_config *cfg = dev->config;
+	int32_t mv = data->raw;
+
+	err = adc_raw_to_millivolts(adc_ref_internal(cfg->adc), cfg->ch_cfg.gain,
+				    cfg->adc_seq.resolution, &mv);
+	if (err) {
+		return err;
+	}
+
+	/* LM35 computes linearly 10mV per 1 degree C */
+	val->val1 = (int32_t)mv / 10;
+	val->val2 = (int32_t)mv % 10;
```

Comment:
Cast is unnecessary (already int32_t), also scaling for val2 is incorrect.

```suggestion
	val->val1 = mv / 10;
	val->val2 = (mv % 10) * 100000;
```

Reply:
Done. One question about the scaling (for my future reference), is this amount of digits expected for the `val2` at sensors? After checking your comment I saw a few other drivers that do this scaling, but didn't found this documented in the docs